File: src/backend/appointment-booking/src/main/java/com/appointment/booking/model/Student.java
package com.appointment.booking.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;
import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@Entity
@Table(name = "student")
public class Student extends User {

    private Date yearOfstudy;
    @OneToOne
    @JoinColumn(name = "school_id", referencedColumnName = "id")
    private School school;

    @OneToMany(mappedBy = "id", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<Session> sessions = new HashSet<>();




}

File: src/frontend/ab-web-app-ui/src/app/core/models/Student.ts
import {User} from "./User";
import {SchoolType} from "./SchoolType";
import {SchoolYear} from "./SchoolYear";
import {SessionDto} from "./session";

export interface Student extends User{
  schoolType?:SchoolType,
  schoolYear?:SchoolYear,
  sessions?:SessionDto[]
}
